### ISSUE TITLE
Instructions for building and publishing with Github Actions to Github Pages

### ISSUE THREAD

**Issue by bmann** [peterkaminski/massivewikibuilder#46]
I'd like to see instructions for GH Actions, to reduce the number of external dependencies e.g Netlify.

This should also be easily adaptable to Gitlab and various other systems that have build and static pages.

I may "answer" this myself, as I have done something similar for TiddlyWiki and LogSeq, but if someone else who is more familiar with Python and GH Actions gets there first, I won't be disappointed :)

**Comment by peterkaminski** [comment]
nice idea, Boris!  lmk if you'd like to pair on GH Actions.

**Comment by bmann** [comment]
Example of the GH Action that builds TWs https://github.com/bmann/tiddlywiki-static-publish/blob/main/.github/workflows/publish.yml

The build.sh has the brief commands, which look a lot like the mw one liner.

**Comment by peterkaminski** [comment]
For netlify deploys, we have people start their wiki using [Massive Wiki Starter](https://github.com/Massive-Wiki/massive-wiki-starter/) as a template.

We probably want that here, too.

I've got a skeleton build set up, the next part is hydrating the submodule for Massive Wiki Builder...

**Comment by peterkaminski** [comment]
@bmann, thanks for the TW example. I've expanded it for Python and MWB.  Check this repo:

<https://github.com/peterkaminski/massive-wiki-for-github-actions>

And the `actions.yml` file:

<https://github.com/peterkaminski/massive-wiki-for-github-actions/blob/main/.github/workflows/actions.yml>

It builds and does a ghpages deploy to somewhere, but I must be missing something, because the GH Pages site doesn't appear where I think it should (currently generates a 404):

https://peterkaminski.github.io/massive-wiki-for-github-actions/

I've got Settings > Pages > Source set to "GitHub Actions", but still no joy.

Will take a break now and come back and read more docs later.

**Comment by peterkaminski** [comment]
> https://peterkaminski.github.io/massive-wiki-for-github-actions/

I switched the Pages source to the `gh-pages` branch, and now the site loads.

The CSS is broken, because it's set up to be at the root, and GitHub Pages deploys to a directory named after the repo.  I can fix that by repointing the hrefs, although that'll make the local theme files slightly different than the others, which would be nice to fix some day.

**Comment by darachm** [comment]
To deploy on gitlab I added in a command:
`find ../../public -regex '.*.html' | xargs sed -i 's/href="\//href="\/basenameOfTheRepo\//g'`
to just change the root reference of the links after the fact
¯\_(ツ)_/¯

**Comment by peterkaminski** [comment]
@darachm, yeah, that's one way to do it! :-)

I think the search would work after another sed, looking for `src="/` rather than `href="/`.

The lines that load the search indexes look like this (with varying numbers):

```html
<script src="/lunr-index-1685993490.5244048.js"></script>
<script src="/lunr-posts-1685993490.5244048.js"></script>
```

**Comment by darachm** [comment]
Yep, the trouble is the links it returns from search don't seem to have a prefix and I don't think you can sed-patch the JS if you're loading it live, not simply :shrug:

**Comment by band** [comment]
Building with Github Actions to Github Pages is implemented and working on an experimental workbench.
Next steps are to (1) create a branch to hold the code and then (2) review and merge that branch.
Implementation date for these steps is TBD.
